pkg/renovate/job.go

```diff
@@ -112,6 +201,8 @@ func (j *JobCoordinator) Execute(ctx context.Context, tasks []*Task) error {
 		return nil
 	}
 
+	registry_secret, err := j.createMergedPullSecret(ctx)
```

Comment:
This secret will live there forever, we need to set this secret to be owned by the job, then when the job is deleted automatically, this secret will be deleted as well.

For example: https://github.com/konflux-ci/mintmaker/blob/8dde1c20a51bca9b74dabb42d7defc703b4432bd/pkg/renovate/job.go#L232-L237